engine/src/main/java/org/hibernate/search/engine/impl/MutableSearchFactoryState.java

```diff
@@ -41,9 +42,9 @@
 	private IndexingMode indexingMode;
 	private Worker worker;
 	private BackendQueueProcessor backendQueueProcessor;
-	private Map<String, FilterDef> filterDefinitions;
+	private Map<String, FilterDef> filterDefinitions = new ConcurrentHashMap<>();
```

Comment:
I would declare all the map members of type `ConcurrentMap` to make their nature apparent. Callers I would just let pass plain maps whose contents I'd put to the internally used concurrent maps. I.e. would not rely on the fact that a concurrent map is passed in (e.g. see factoryState.setDocumentBuildersIndexedEntities(...) further below in this PR).